### ISSUE TITLE
Movit:  Kdenlive rendering  to screen unless --avformat-no-vdpau used or GPL3 not used

### ISSUE THREAD

**Issue by lukefromdc** [mltframework/mlt#106]
If MLT 6.1.0 is built with --enable-gpl3 , used with Kdenlive, and Kdenlive is set up with Movit enabled, all render jobs get diverted to a window on the screen. The output file remains at 0 bytes. Adding  --avformat-no-vdpau to the configuration options stops this, as does buidling without GPLv3 support. I had to read the configure script  for the avformat module to find this and sucessfully build for Movit with GPL3 components (video quality measurement transition) enabled. It was not listed anywhere in the options displayed by` ./configure --help ` and it should be. That line and a note that it is needed for kdenlive would save other builders a lot of trouble. In short, this is a documentation issue-the code works. Only those builders using GPLv3 will encounter this, for me the render output diversion was a mystery that took months to figure out.

**Comment by ddennedy** [comment]
> It was not listed anywhere in the options displayed by./configure --help
What is "it?"
I really doubt --enable-gpl3 has anything to do with it. I think you are just flailing around with the build, and stumbled upon something to make it work, but don't know what.

**Comment by ddennedy** [comment]
Also, "--avformat-no-vdpau" basically does nothing. It sets a var to false that is already false by default (unless you also used "--avformat-vdpau").

**Comment by ddennedy** [comment]
"--enable-gpu3" enables the vqm transition, which has nothing to do with this. And if it does, that is possibly Kdenlive's fault, not MLT. MLT does not automatically add this transition anywhere to compositions. It's existence does not interfere. It works fine in Shotcut, which includes it.

**Comment by lukefromdc** [comment]
Closing as no longer needed -disable-qt may have been the issue here, it was the difference between qglsl showing up and not showing up in the list of consumers and in where the rendering went. Still experimenting...

**Comment by ddennedy** [comment]
> disable-qt may have been the issue here, it was the difference between qglsl showing up

Now, that does make sense! Qt is used to obtain an OpenGL context in a cross-platform manner to supply to Movit.
FYI, docs about OpenGL and Movit in MLT are documented here, which is linked from the main documentation page on the site: https://www.mltframework.org/bin/view/MLT/OpenGL

**Comment by lukefromdc** [comment]
When I got different results tonight than I got with 0.9.9 with the same 
build options I simply had to experiment and find out what worked. I had
orignally thought VDPAU was behind this mess, had you not told me about
the QT module providing the OpenGL context I would still not know better.

Sorry for any hassle, I use Kdenlive with Movit and was stumped for months
by this issue. I think the only thing -enable-gpl3 had to do with it was it was
part of the same string of options I decided to simply cut off and see if the 
trouble was in it. The orignal option string I used to build MLT for kdenlive
15.04 (which did not support Movit) was 

./configure --prefix=/usr --enable-gpl --enable-gpl3 --disable-qt ---avformat-vdpau

As I was experimenting with VDPAU to see if it would have any performance
benefits (it did not) and builds with KDE4 on the system as well as KF5 at that
time build against QT4 and segfaulted Kdenlive. Maybe that was one of the 
reasons Movit was removed until 15.08? Anyway, I saw no guidance from anywhere
to re-enable the QT module, and when I tried to find out what was diverting render 
jobs to the screen I was unable to find any information at all about what was going
on. Thus months of trial and error on my part, though I still did get Movit working 
months before Debian started shipping MLT with it as a dependency.
